src/views/SalesAdd.vue

```diff
@@ -0,0 +1,200 @@
+const calendarDialog = ref(false);
+const [isProductSelected, isDateSelected] = [ref(false), ref(false)];
+const formattedDate = ref("");
+
+const allUsers = computed(() => store.getters["users/allUsers"]);
+const allUsersNames = allUsers.value.map((user) => user.name);
+
+watch(seller, (newValue) => {
+  selectedUser.value = allUsers.value.find((user) => user.name == seller.value);
+  productsContent.value = [];
+});
+
+const totalAmount = computed(() =>
+  productsContent.value.reduce((acc, cur) => acc + Number(cur.salePrice), 0)
+);
+
+const discountedSmallAmount = computed(() =>
+  productsContent.value.reduce(
+    (acc, cur) => acc + (cur.salePrice * (cur.discount ?? 0)) / 100,
+    0
+  )
+);
+
+const discountedAmount = computed(
+  () => totalAmount.value - discountedSmallAmount.value
+);
+
+const totalDiscount = computed(() => {
+  const totalAmountValue = totalAmount.value;
+  const discountedAmountValue = discountedAmount.value;
+
+  if (totalAmountValue === 0 || isNaN(discountedAmountValue)) {
+    return 0;
+  }
+
+  return 100 - (discountedAmountValue / totalAmountValue) * 100;
+});
+
+const toggleDialog = (isOpen) => {
+  productsDialog.value = isOpen;
+};
+
+function handleCloseCalendar(isOpen, selectedDate) {
+  calendarDialog.value = isOpen;
+  formattedDate.value = selectedDate;
+  isDateSelected.value = true;
+}
+
+const productsTableValues = (productsContentValue) => {
+  productsContent.value = productsContentValue;
+  toggleDialog(false);
+};
+
+const resetForm = () => {
+  if (form.value) {
+    form.value.reset();
+    form.value.resetValidation();
+    seller.value = [];
+    buyer.value = [];
+    productsContent.value = [];
+    isProductSelected.value = false;
+    isDateSelected.value = false;
+  }
+};
+
+const handleSubmit = async () => {
+  const { valid } = await form.value.validate();
+
+  if (!valid) return;
+
+  if (productsContent.value.length <= 0) {
+    snackbarProvider.showErrorSnackbar("Please select a product!");
+    return;
+  }
+
+  if (formattedDate.value.length <= 0) {
+    snackbarProvider.showErrorSnackbar("Please select a date!");
+    return;
+  }
+
+  const getUserById = (username) =>
+    allUsers.value.find((user) => user.name === username).id;
+
+  const selectedSeller = getUserById(seller.value);
+  const selectedBuyer = getUserById(buyer.value);
+
+  const selectedProducts = productsContent.value.map((product) => ({
+    productId: product.id,
+    salePrice: Number(product.salePrice),
+    discount: Number(product.discount) || 0,
+  }));
+
+  const data = {
+    sellerId: selectedSeller,
+    buyerId: selectedBuyer,
+    products: selectedProducts,
+    date: formattedDate.value,
+  };
+
+  try {
+    await store.dispatch("sales/postSale", data);
+    snackbarProvider.showSuccessSnackbar("Successfully sold the product!");
+    router.push("/sales");
+  } catch (error) {
+    snackbarProvider.showErrorSnackbar("Couldn't sale the product");
+  }
+};
```

Comment:
Can you try splitting this method into smaller ones? I suggest separating into validation checks, data preparation, sending request to backend

Reply:
done